persist/fs/seek.go

```diff
@@ -320,63 +339,53 @@ func (s *seeker) SeekByID(id ts.ID) (checked.Bytes, error) {
 // instead of looking it up on its own. Useful in cases where you've already
 // obtained an entry and don't want to waste resources looking it up again.
 func (s *seeker) SeekByIndexEntry(entry IndexEntry) (checked.Bytes, error) {
-	_, err := s.dataFd.Seek(entry.Offset, 0)
-	if err != nil {
-		return nil, err
+	// Should never happen, but prevent panics if somehow we're provided an index entry
+	// with a negative or too large offset
+	if int(entry.Offset) > len(s.dataMmap)-1 {
+		return nil, errInvalidDataFileOffset
 	}
-	s.dataReader.Reset(s.dataFd)
 
-	n, err := s.dataReader.Read(s.prologue)
-	if err != nil {
-		return nil, err
-	} else if n != cap(s.prologue) {
+	// We'll treat the dataView similar to a reader interface in that we will continue to reslice
+	// it such that the first byte is always the next byte, and we'll focus instead on how much
+	// we want to read.
+	dataView := s.dataMmap[entry.Offset:]
+
+	// Should never happen, but prevents panics in the case of malformed data
+	if len(dataView) < cap(s.prologue) {
```

Comment:
Hm, do we need `s.prologue` any more?  Maybe we can remove this field?

Reply:
is it 

```golang
len(data) < markerLen
```

or

```golang
len(data) < markerLen + idxLen
```